### DISCUSSION TITLE
Variable height within top layers (like scarf joints) to make slope transitions smoother.

### DISCUSSION THREAD

**Discussion by jpettitt** [OrcaSlicer/OrcaSlicer#4918] [Ideas]
**tl;dr use scarf joint approach to make layer step on top surfaces smaller.**

One repeated and annoying quality issue is long low angle height changes are sliced in a model.  For example this part at 0.2 layer height looks ugly.

<img width="640" alt="image" src="https://github.com/SoftFever/OrcaSlicer/assets/524856/a5f29f85-c301-4856-bf41-449e832ccd29">

Changing to 0.08 layers or using adaptive layers helps but does not solve the issue

<img width="640" alt="image" src="https://github.com/SoftFever/OrcaSlicer/assets/524856/3747b7ed-670f-426e-99aa-2c4701159ffa">

I occurred to me that scarf joints use a variable height within a layer to introduce many more, much smaller, steps.  Could this same approach be applied to top layers?  Taking the example part above and the default 10 scarf steps per layer with would give 10x the number of surface steps and greatly smooth the finish of the part.

It would also be really good at curved to flat surface transitions which never really look good on FDM
